### PR TITLE
Improve robustness of collection edit E2E test.

### PR DESCRIPTION
**Pull request by jmchilton**
#16099 hopefully fixed the underlying error causing the flakiness in the test but I still have a bunch of refactorings here aimed at using improved abstractions, targeting more explicit, stable DOM elements, etc.

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy_test/selenium/test_collection_edit.py [hunk 1]
```diff
@@ -42,64 +43,58 @@ def test_change_datatype_simple_list(self):
...
-        self._collection_dropdown("build list")
-        self.collection_builder_set_name("my cool list")
-        self.screenshot("collection_builder_list")
-        self.collection_builder_create()
-        self._wait_for_hid_visible(2)
 
-    def create_simple_list_collection_txt(self):
-        self.perform_upload(self.get_filename("1.txt"))
+        @retry_assertion_during_transitions
+        def check_dataset_converted():
```
Comment (mvdbeek) [inline lib/galaxy_test/selenium/test_collection_edit.py]:
```suggestion
        def check_dataset_changed():
```
the conversion would always create a new HDA, so I first thought `self.history_panel_item_component(hid=1)` was the problem

File: lib/galaxy_test/selenium/test_collection_edit.py [hunk 2]
```diff
@@ -42,64 +43,58 @@ def test_change_datatype_simple_list(self):
...
+            # there is some issue with the cache of dataset details in the client
```
Comment (mvdbeek) [inline lib/galaxy_test/selenium/test_collection_edit.py]: I think I know what the problem is, the bulk update endpoint this is targeting doesn't update the HDCA update_time, so the history doesn't know that anything changed. I'm working on a fix for that.

File: lib/galaxy_test/selenium/test_collection_edit.py [hunk 3]
```diff
@@ -42,64 +43,58 @@ def test_change_datatype_simple_list(self):
...
+            # not being updated for reflect this change - the retry and home here
+            # are meant to work around that.
+            self.home()
+            self.history_panel_expand_collection(2)
+            self.history_panel_ensure_showing_item_details(1)
+            item = self.history_panel_item_component(hid=1)
+            item.datatype.wait_for_visible()
+            assert item.datatype.wait_for_text() == dataNew
+
+        check_dataset_converted()
```
Comment (mvdbeek) [inline lib/galaxy_test/selenium/test_collection_edit.py]:
```suggestion
        check_dataset_changed()
```

**Review by mvdbeek — COMMENTED** [review]
Nice!

File: lib/galaxy_test/selenium/test_collection_edit.py (hunk 2 above)
Comment (mvdbeek) [reply lib/galaxy_test/selenium/test_collection_edit.py]: https://github.com/galaxyproject/galaxy/pull/16099 should fix this

Comment (jmchilton) [reply lib/galaxy_test/selenium/test_collection_edit.py]: I'll redo this without the cache busting then.

Comment (mvdbeek) [reply lib/galaxy_test/selenium/test_collection_edit.py]: thank you! I do hope this was really it 🤞

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy_test/selenium/test_collection_edit.py (hunk 2 above)
Comment (jmchilton) [reply lib/galaxy_test/selenium/test_collection_edit.py]: I've seen this two or three times reviewing PRs this morning. I think this didn't work? Should I redo the cache busting or do you have another idea?

Comment (mvdbeek) [reply lib/galaxy_test/selenium/test_collection_edit.py]: I would prefer to fix it if possible. Do you have an action link still ?

Comment (jmchilton) [reply lib/galaxy_test/selenium/test_collection_edit.py]: https://github.com/galaxyproject/galaxy/actions/runs/5136253645/jobs/9242755222#step:11:5292

Here is the screenshot I downloaded for one of the runs - I am not sure if it was this one (probably it wasn't) but the stack trace was the same:

last

Before I had verified the bug manually - I could try that again and verify the test still makes sense.

Comment (mvdbeek) [reply lib/galaxy_test/selenium/test_collection_edit.py]: Hmm, another problem I ignored before was that `self.wait_for_history()` doesn't do anything here, we're not launching a job. Maybe a small sleep would fix this ? I would really like to assure that the collection item gets updated without a reload.

**Pull request merged**